src/TestEditor.tsx

```diff
@@ -1,12 +1,17 @@
 import { type ReactElement } from 'react';
-import type { Test, TestInputs } from './generated/test_case';
+import type { Test, TestInputs, TestRunResults } from './generated/test_case';
 import TestInputsEditor from './TestInputsEditor';
 import TestOutputsEditor from './TestOutputsEditor';
 
 type Props = {
   test: Test;
   onTestChange(newValue: Test): void;
   onTestDelete(testScope: string): void;
+  onTestRun(testScope: string): void;
+  runState?: {
+    status: 'running' | 'success' | 'error';
+    results?: TestRunResults;
```

Comment:
I don't know how you implement this, but I guess every time you modify the test the running state should be reset (because the test result might change).